### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by tobiasake**
Dockerfile for building Eiffel-Intelligence Backend docker image.

Eiffel-Intelligence Backend image is built with this command. Uses EI version from Jitpack:
`docker build -f src/main/docker/Dockerfile -t eiffel-intelligence-backend:0.1 .`

Run EI image/container with following command:
`docker run -p 8034:8091 --expose 8091 --network=eiffel2_eiffel_2.0_1 -e server.port=8091 -e logging.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e logging.level.com.ericsson.ei=DEBUG eiffel-intelligence-backend:0.1`

With following command example you should be able to access EI RestApi interface wiith address:
localhost:8034/(RestApi end points) or (docker-host ip address):8034/(RestApi end points)

Wiki page will be updated with this information when this has been merged.

### CODE REVIEW

**Comment by tobiasake** [comment]
Shall I change dockerfile name? From "DockerfileUseLocalCode" to DockerfileUseLocalWarFile" or "DockerfileUseLocalVersion" ? or something else.

**Comment by emichaf** [comment]
create folder under src/main/docker
and add file: Dockerfile

File: wiki/markdown/docker.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Docker
+
+In Eiffel-Intelligence source code a Dockerfile is provided which helps the developer or user to build the local Eiffel-Intellegence source code changes to a Docker image.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: I think it sounds more natural to have "source code repository"

File: wiki/markdown/docker.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+With the Docker image user can tryout the Eiffel-Intelligence on a Docker Host or in a Kubernetes cluster.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: change tryout -> try out 
or to "try-out" according to the cambridge dictionary :)

File: wiki/markdown/docker.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  Windows: https://docs.docker.com/docker-for-windows/install/
+
+## Follow these step to build the Docker image.
+
+1. Build the Eiffel-intelligence war file: 
+`mvn package -DskipTests` 
+
+
+This will produce a war file in the "target" folder.
+2. Build the Docker image with the war file that was produced from previous step:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: Have number 2 start on a new line for more readability

File: wiki/markdown/docker.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+It is possible to set all Spring available properties via docker envrionment "-e" flag. See the application.properties file for all available Eiffel-Intelligence Spring properties.
+
+
+Docker flags:
+
+
+"--expose 8091" - this Docker flag tells that containers internal port shall be exposed to outside of the Docker Host. This flag do not set which port that should be allocated outside Docker Host on the acual server/machine.
+
+
+"-p 8034:8091" - this Docker flag is mapping the containers exposed port to a port number that will be allocated outside Docker host. 8091 is the exposed port in the container. 8034 is the port number that port number that will be allocated outside Docker host and mapped to the Docker Host internal exposed port number 8091.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: "8034 is the port number that port number that will be allocated outside Docker" Too many port number in this sentence. It reads very confusing.

File: wiki/markdown/docker.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+`docker run -p 8034:8091 --expose 8091 -e server.port=8091 -e logging.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e logging.level.com.ericsson.ei=DEBUG eiffel-intelligence-backend:0.1`
+
+Some info of all flags to this command: 
+
+
+Eiffel Intelligence Spring Properties:
+
+
+"-e server.port=8091" - Is the Spring property setting for Eiffel-Intelligence applications web port.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: Try to highlight all the flags in bold or in small code snippets to make them easier to find in the text.

File: wiki/markdown/docker.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+logging.level.com.ericsson.ei=DEBUG" - These Spring properties set the logging level for the Eiffel-Intelligence applications. 
+
+
+It is possible to set all Spring available properties via docker envrionment "-e" flag. See the application.properties file for all available Eiffel-Intelligence Spring properties.
+
+
+Docker flags:
+
+
+"--expose 8091" - this Docker flag tells that containers internal port shall be exposed to outside of the Docker Host. This flag do not set which port that should be allocated outside Docker Host on the acual server/machine.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: acual -> actual

File: wiki/markdown/docker.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+
+"-e server.port=8091" - Is the Spring property setting for Eiffel-Intelligence applications web port.
+
+
+"-e logging.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e 
+logging.level.com.ericsson.ei=DEBUG" - These Spring properties set the logging level for the Eiffel-Intelligence applications. 
+
+
+It is possible to set all Spring available properties via docker envrionment "-e" flag. See the application.properties file for all available Eiffel-Intelligence Spring properties.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: You could add a link to the application.properties file

File: wiki/markdown/docker.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+## Run Docker image on local Docker Host
+To run the produced docker image on the local Docker host, execute this command: 
+
+
+`docker run -p 8034:8091 --expose 8091 -e server.port=8091 -e logging.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e logging.level.com.ericsson.ei=DEBUG eiffel-intelligence-backend:0.1`
+
+Some info of all flags to this command: 
+
+
+Eiffel Intelligence Spring Properties:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: Make this a small header.

File: wiki/markdown/docker.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+
+"-e logging.level.root=DEBUG -e logging.level.org.springframework.web=DEBUG -e 
+logging.level.com.ericsson.ei=DEBUG" - These Spring properties set the logging level for the Eiffel-Intelligence applications. 
+
+
+It is possible to set all Spring available properties via docker envrionment "-e" flag. See the application.properties file for all available Eiffel-Intelligence Spring properties.
+
+
+Docker flags:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/docker.md]: Same comment here. Make this a small header.

File: wiki/markdown/docker.md (hunk 1 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 2 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 4 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Re-wrote this section a bit.

File: wiki/markdown/docker.md (hunk 5 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 9 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 8 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 7 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

File: wiki/markdown/docker.md (hunk 6 above)
Comment (tobiasake) [reply wiki/markdown/docker.md]: Fixed.

**Pull request merged**